src/backend/src/services/teams.services.ts

```diff
@@ -284,4 +284,27 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  static async archiveTeam(user: User, teamId: string): Promise<Team> {
+    const team = await prisma.team.findFirst({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
+    if (!isAdmin(user.role)) {
+      throw new AccessDeniedException('You must be an admin or above to archive a team');
+    }
+
+    if (team.projects.some((project) => project.wbsElement.status !== WBS_Element_Status.COMPLETE)) {
+      throw new HttpException(400, 'A team is not archivable if it has any active projects, or incomplete projects');
+    }
+
+    team.userArchivedId = user.userId;
+
+    team.dateArchived = team.dateArchived !== null ? new Date() : null;
+
+    return teamTransformer(team);
+  }
```

Comment:
It seems like you are editing the variable team, but really updating it. To update the database, you will have to call prisma.team.update(...)